lib/posix/CMakeLists.txt

```diff
@@ -58,6 +58,7 @@ zephyr_library_sources_ifdef(CONFIG_PTHREAD_RWLOCK rwlock.c)
 zephyr_library_sources_ifdef(CONFIG_POSIX_PRIORITY_SCHEDULING sched.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC semaphore.c)
 zephyr_library_sources_ifdef(CONFIG_PTHREAD_SPINLOCK spinlock.c)
+zephyr_library_sources_ifdef(CONFIG_POSIX_PUTMSG stropts.c)
```

Comment:
would it make more sense for this to use something like `CONFIG_POSIX_STROPTS` instead?

Reply:
Absolutely, since STREAMS has other methods. Made this change!

Reply:
I asked for putmsg originally

The reason is that we want to have Kconfig automatically create the correct implementation conformance macros for us if all of the functions in a specific option group or option are selected.

Reply:
Should I revert to using putmsg?

Reply:
A question re: that, when I implement functions like putpmsg() later, their config symbols would just replace the PUTMSG part, right?

Reply:
Just add a new Kconfig per function.

It will make more sense once we have the PSE shortcut options added

Reply:
@cfriedt I added your commits re: `zassert_not_equal(RS_HIPRI, ~RS_HIPRI);`, but it seems we don't need it since we're not dealing with priorities all together and these are just stubs. Yong earlier suggested I remove it. Wondering which way to go.

Reply:
It is needed for implementation conformance i.e. so that an application is able to compile.